modules/core/src/main/java/org/apache/synapse/deployers/LocalEntryDeployer.java

```diff
@@ -79,6 +102,66 @@ public String deploySynapseArtifact(OMElement artifactConfig, String fileName,
         return null;
     }
 
+    private void handleHttpConnectorCertificates(OMElement element) throws DeploymentException {
+
+        OMElement httpInitElement =
+                element.getFirstChildWithName(new QName(SynapseConstants.SYNAPSE_NAMESPACE, HTTP_CONNECTION_IDENTIFIER));
+        if (httpInitElement != null) {
+            Iterator childElementIterator = httpInitElement.getChildElements();
+            while (childElementIterator.hasNext()) {
+                OMElement childElement = (OMElement) childElementIterator.next();
+                String childElementValue = childElement.getText();
+                String transformedElementValue = getTransformedElementValue(childElementValue);
+                if (transformedElementValue.endsWith(CERTIFICATE_EXTENSION)) {
+                    loadCertificateFileToStore(transformedElementValue);
+                }
+            }
+        }
+    }
+
+    private void loadCertificateFileToStore(String certificateFileResourceKey) throws DeploymentException {
+
+        String certificateFilePath = getSynapseConfiguration().getRegistry().getRegistryEntry(certificateFileResourceKey).getName();
+        File certificateFile = new File(certificateFilePath);
+        String certificateAlias = certificateFile.getName().split("\\.")[0];
+        try {
+            FileInputStream certificateFileInputStream = FileUtils.openInputStream(new File(certificateFilePath));
+            SslSenderTrustStoreHolder sslSenderTrustStoreHolder = SslSenderTrustStoreHolder.getInstance();
+            KeyStore sslSenderTrustStore = sslSenderTrustStoreHolder.getKeyStore();
+            CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+            Certificate certificate = certificateFactory.generateCertificate(certificateFileInputStream);
+            sslSenderTrustStore.setCertificateEntry(certificateAlias, certificate);
+
+            FileOutputStream fileOutputStream = new FileOutputStream(sslSenderTrustStoreHolder.getLocation());
+            sslSenderTrustStore.store(fileOutputStream, sslSenderTrustStoreHolder.getPassword().toCharArray());
+
+            FileInputStream fileInputStream = new FileInputStream(sslSenderTrustStoreHolder.getLocation());
+            InputStream dest = IOUtils.toBufferedInputStream(fileInputStream);
+            fileInputStream.close();
+            sslSenderTrustStore.load(dest, sslSenderTrustStoreHolder.getPassword().toCharArray());
```

Comment:
Do we need these redundant truststore loads as we do in line 145?

Reply:
Yes. They represent 2 things. First load is to load the certificate to the JKS and the second load is used to load the new JKS and create a new connection.